src/lib/indexer/index.ts

```diff
@@ -254,4 +273,41 @@ export default class Indexer {
       }
     })
   }
+  getStatus = async (): Promise<Status> => {
+    const currentDate = new Date()
+    if (currentDate.getTime() - this.startupTime.getTime() < 30 * 1000) {
+      // if we started less than 30s ago -> PASS
+      return {
+        status: serviceState.UP,
+        detail: { message: 'Service healthy.' },
+      }
+    } else {
+      // we started more than 30s ago
+      if (this.lastProcessedBlockTime !== null) {
+        // if we have already started processing blocks check if the last was within 30s
+        const differenceInMilliseconds = currentDate.getTime() - this.lastProcessedBlockTime.getTime()
+        // if isMoreThan30Seconds is true -> will be unhealthy
+        if (differenceInMilliseconds > 30 * 1000) {
+          return {
+            status: serviceState.DOWN,
+            detail: { message: 'Last processed block was more than 30s ago.' },
+          }
+        }
+      } else if (this.lastUnprocessedBlockTime !== null) {
+        // if we are still catching up on old blocks check if last was within 30s
+        const differenceInMilliseconds = currentDate.getTime() - this.lastUnprocessedBlockTime.getTime()
+        // if isMoreThan30Seconds is true -> will be unhealthy
+        if (differenceInMilliseconds > 30 * 1000) {
+          return {
+            status: serviceState.DOWN,
+            detail: { message: 'Started more than 30s ago and no blocks have been processed or registered.' },
+          }
+        }
+      }
+    }
+    return {
+      status: serviceState.UP,
+      detail: { message: 'Service healthy.' },
+    }
+  }
```

Comment:
I think the logic here can be simplified to make it easier to reason about. It looks to me like there's a bug here if both times are null. 

Try to avoid `else` statements and instead rely on return statements in an `if` block. The order I think I'd consider is as follows:

1. if `startup` was less than `30s` ago then return `UP` 
2. if `lastProcessedBlockTime` and `lastUnprocessedBlockTime` are both `null` return `DOWN` (true because we know startup was longer than 30s ago)
3. consider the `latestActivityTime` as  `lastProcessedBlockTime || lastUnprocessedBlockTime`. (We know one must be truthy)
4. if `lastActivityTime` was less than `30s` ago return `UP`
5. return `DOWN` with the message derived from which one of the dates is truthy